src/core/types.ts

```diff
@@ -92,8 +92,8 @@ export class Args<T extends Variable[]> {
 }
 
 export class Triggers {
-    list: Trigger[];
-    constructor(trigger: Trigger, ...triggers: Trigger[]) {
+    list: Array<Trigger | Array<Trigger>>;
+    constructor(trigger: Trigger | Array<Trigger>, ...triggers: Array<Trigger | Array<Trigger>>) {
```

Comment:
With this change, I think it's also necessary to handle the `Array<Trigger>` case in `addReaction`/`_recordDeps` since we those not match the `instanceof Trigger` type guards.

Reply:
Good points, thanks! I think I addressed your points here: https://github.com/lf-lang/reactor-ts/pull/89/commits/8c6d4b52f5b837452858a21adc7cf254d35fdf09

Please take another look!

Reply:
Looks good!